src/radical/pilot/pilot_manager.py

```diff
@@ -77,33 +77,47 @@ def __init__(self, session):
 
         **Returns:**
             * A new `PilotManager` object [:class:`radical.pilot.PilotManager`].
-        """
+        '''
+
+        assert(session.primary), 'pmgr needs primary session'
 
-        self._bridges     = dict()
-        self._components  = dict()
+        self._session     = session
         self._pilots      = dict()
         self._pilots_lock = mt.RLock()
         self._callbacks   = dict()
         self._pcb_lock    = mt.RLock()
         self._terminate   = mt.Event()
         self._closed      = False
         self._rec_id      = 0       # used for session recording
+        self._uid         = ru.generate_id('pmgr.%(item_counter)04d',
+                                           ru.ID_CUSTOM, ns=session.uid)
 
         for m in rpc.PMGR_METRICS:
             self._callbacks[m] = dict()
 
-        cfg = ru.read_json("%s/configs/pmgr_%s.json" \
-                % (os.path.dirname(__file__),
-                   os.environ.get('RADICAL_PILOT_PMGR_CFG', 'default')))
-
-        assert(cfg['db_poll_sleeptime']), 'db_poll_sleeptime not configured'
-
-        # initialize the base class (with no intent to fork)
-        self._uid    = ru.generate_id('pmgr.%(item_counter)04d', ru.ID_CUSTOM,
-                                      namespace=session.uid)
-        cfg['owner'] = self.uid
-        rpu.Component.__init__(self, cfg, session)
-        self.start(spawn=False)
+        # NOTE: `name` and `cfg` are overloaded, the user cannot point to
+        #       a predefined config and amed it at the same time.  This might
+        #       be ok for the session, but introduces a minor API inconsistency.
+        #
+        name = None
+        if isinstance(cfg, str):
+            name = cfg
+            cfg  = None
+
+        cfg           = ru.Config('radical.pilot.pmgr', name=name, cfg=cfg)
+        cfg.uid       = self._uid
+        cfg.owner     = self._uid
+        cfg.sid       = self._session.uid
```

Comment:
For consistency, should this be session.sid?

Reply:
See above, this should be ok.